### PR TITLE
fix(setting): create a real SC for the setting "default-longhorn-static-storage-class"

### PR DESCRIPTION
**Pull request by mantissahz**
#### Which issue(s) this PR fixes:

Issue # longhorn/longhorn#6446

### CODE REVIEW

File: controller/setting_controller.go [hunk 1]
```diff
@@ -965,6 +970,34 @@ func (sc *SettingController) updateLogLevel() error {
...
+	_, err = sc.ds.GetStorageClassRO(defaultStaticStorageClassName)
+	if err != nil && apierrors.IsNotFound(err) {
+		allowVolumeExpansion := true
+		storageClass := &storagev1.StorageClass{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: defaultStaticStorageClassName,
+			},
+			Provisioner:          types.LonghornDriverName,
+			AllowVolumeExpansion: &allowVolumeExpansion,
+			Parameters:           map[string]string{"staleReplicaTimeout": "30"},
```
Comment (derekbit) [inline controller/setting_controller.go:989]: Only 30 seconds?

Comment (mantissahz) [reply controller/setting_controller.go:989]: That's 30 minutes. https://github.com/longhorn/longhorn/issues/6446#issuecomment-1861816858

**Review by derekbit — APPROVED** [review]
LGTM.

Also need to write a KB about a workaround for users with PVC with longhorn-static SC in older version.

**Comment by innobead** [comment]
@mergify backport v1.6.x v1.5.x

**Pull request merged**